SwiftReflector/CompilationSettings.cs

```diff
@@ -0,0 +1,224 @@
+			TargetRepresentation = Exceptions.ThrowOnNull (targetRepresentation, nameof (targetRepresentation));
+		}
+
+		public string OutputDirectory { get; private set; }
+		public string ModuleName { get; private set; }
+		public List<string> FrameworkPaths { get; private set; }
+		public List<string> LibraryPaths { get; private set; }
+		public List<string> SwiftFilePaths { get; private set; }
+		public UniformTargetRepresentation TargetRepresentation { get; private set; }
+		public string MakeFrameworkScript { get; private set; }
+		public string WorkingDirectory { get; private set; }
+
+
+		public string CompileTarget ()
+		{
+			var scriptPath = GetScriptPath ();
+			var args = BuildCommandArgs ();
+			var output = ExecAndCollect.Run (scriptPath, args, WorkingDirectory);
+			if (TargetRepresentation.Library != null) {
+				var frameworkDir = Path.Combine (OutputDirectory, $"{ModuleName}.framework");
+				var libraryPath = Path.Combine (OutputDirectory, $"lib{ModuleName}.dylib");
+				File.Copy (Path.Combine (frameworkDir, ModuleName), libraryPath);
+				Directories.DeleteContentsAndSelf (frameworkDir);
+				ChangeInstallName (libraryPath);
+			}
+			return output;
+		}
+
+		void ChangeInstallName (string libraryPath)
+		{
+			var args = $"install_name_tool -id @rpath/lib{ModuleName}.dylib {libraryPath}";
+			var output = ExecAndCollect.Run ("xcrun", args);
+		}
+
+		string BuildCommandArgs ()
+		{
+			// Here are the args to the script that we need:
+			//--frameworks fw1 fw2 ...
+			//   adds framework directories to swift compilation (-F) (optional)
+			//--libraries lb1 lb2...
+			//   adds library directories to swift compilation (-L) (optional)
+			//--swift-files sf1 sf2 ...
+			//   adds swift files to be compiled (required)
+			//--target-os os - name, one of ios, tvos, watchos, macosx (required)
+			//   sets the target operating system for the build.
+			//--minimum-os-version version (required)
+			//   sets the minimum operating system version for the compilation.
+			//--simulator-archs arch1 arch2...
+			//   sets the architectures for a simulator build.
+			//--device-archs arch1 arch2...
+			//   sets the architectures for a device build.
+			//--module-name name (required)
+			//   sets the name of the output module.
+			//--extra-swift-args arg1 arg2...
+			//   sets extra arguments to pass to the swift compiler.
+			//--output-path path (required)
+			//   sets the directory where there final output will live.
+			//--make-xcframework (optional)
+			//   if present, puts both device and simulator builds into an xcframework
+			//   if present, both--simulator - archs and--device - archs must be present.
+
+			if (SwiftFilePaths.Count == 0)
+				throw new Exception ("No files provided to compile.");
+			CheckAllFilesExist ();
+			var args = new StringBuilder ();
+
+			// useful for debugging:
+			// args.Append (" --verbose --verbose ");
+
+			args.Append ("--output-path ").Append (OutputDirectory);
```

Comment:
This won't work if `OutputDirectory` contains a space.

I've found that it's usually much better to keep arguments as an array/list until the very end, and only then convert to a single string:

https://github.com/xamarin/xamarin-macios/blob/d11852e02c3d51be71b6989e0eacb07e945b0e82/tools/common/Execution.cs#L68

Reply:
Agree, but in this case 2 changes fixes them all instead of more than a dozen changes.